includes/payment-methods/class-afterpay-payment-method.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Class Afterpay_Payment_Method
+ *
+ * @package WCPay\Payment_Methods
+ */
+
+namespace WCPay\Payment_Methods;
+
+use WC_Payments_Token_Service;
+
+/**
+ * Afterpay Payment Method class extending UPE base class
+ */
+class Afterpay_Payment_Method extends UPE_Payment_Method {
+
+	const PAYMENT_METHOD_STRIPE_ID = 'afterpay_clearpay';
+
+	/**
+	 * Constructor for link payment method
+	 *
+	 * @param WC_Payments_Token_Service $token_service Token class instance.
+	 */
+	public function __construct( $token_service ) {
+		parent::__construct( $token_service );
+		$this->stripe_id   = self::PAYMENT_METHOD_STRIPE_ID;
+		$this->title       = __( 'Afterpay', 'woocommerce-payments' );
+		$this->is_reusable = false;
+		$this->currencies  = [ 'USD', 'CAD', 'AUD', 'NZD', 'GBP', 'EUR' ];
```

Comment:
I'm getting 
> ERROR The currency provided (eur) is invalid. Payments with afterpay_clearpay support the following currencies: usd. (invalid_request_error)
error unless I leave only USD as currency.

The list provided in this PR seems correct [as per Stripe documentation](https://stripe.com/docs/payments/afterpay-clearpay) 

@mgascam @nagpai

Reply:
What account and settings are you using when you get this error? 

You may have done it, but just to be sure, have you added the db fees using command for the country / currency combination that you are testing? unless that is added, I guess the method wont show up.

Reply:
> Stripe limits the currency that Afterpay can accept payments in based on your Stripe account’s registered location.

From this I conclude that we'll have to set `$currencies` dynamically based from account. Don't know about any side-effects that might pop out

Reply:
In the db fee rule, we specify country as well as currency. eg. Spain would have an Afterpay ( Clearpay ) DB fee rule specified only for Euro. Will that not be sufficient to not show Afterpay, unless the country and currency both dont match?

Reply:
I might have to run further tests. Can't reproduce today anymore...

Reply:
> In the db fee rule, we specify country as well as currency. eg. Spain would have an Afterpay ( Clearpay ) DB fee rule specified only for Euro. Will that not be sufficient to not show Afterpay, unless the country and currency both dont match?

As I recall, code checks for presence of the payment method in the fees, not at the how fee is defined. Thus, if there is any fee for Afterpay, payment method will be visible for any currency. However, checkout block and UPE might have additional checks in place.